.circleci/config.yml

```diff
@@ -0,0 +1,46 @@
+version: 2.1
+
+commands:
+  install-deps:
+    description: Install dependencies using pre-commit
+    steps:
+      - run:
+          command: |
+            pip install pre-commit tox flit
+            pre-commit install
+          name: Install dependencies
+
+jobs:
+  build:
+    docker:
+      - image: circleci/python:3.7
+    steps:
+      - checkout
+      - install-deps
+      - run: tox
+  deploy:
+    docker:
+      - image: circleci/python:3.7
+    steps:
+      - checkout
+      - install-deps
+      - run:
+          command: |
+            FLIT_USERNAME=$PYPI_USERNAME FLIT_PASSWORD=$PYPI_PASSWORD flit publish
```

Comment:
The FLIT_USERNAME and FLIT_PASSWORD should be stored in the CircleCI settings as environment variables. No need to pass them explicitly here. 

Also you need to `flit build` before `flit publish`